crates/proto/src/op/message.rs

```diff
@@ -501,17 +514,21 @@ impl Message {
     /// ```
     /// # Return value
     ///
-    /// Returns the EDNS record if it was found in the additional section.
+    /// Optionally returns a reference to EDNS section
     pub fn edns(&self) -> Option<&Edns> {
         self.edns.as_ref()
     }
 
-    /// If edns is_none, this will create a new default Edns.
-    pub fn edns_mut(&mut self) -> &mut Edns {
+    /// Optionally returns mutable reference to EDNS section
+    pub fn edns_mut(&mut self) -> Option<&mut Edns> {
```

Comment:
What do you think about deprecating the existing methods, and adding new one that would be call `extensions` and `extensions_mut`?

Then we can skip the breaking change and the name would be more explicit.

Reply:
I had a comment I thought I posted but I guess I didn't!

Yeah, after working with the API a bit I totally see how you arrived at the original signature. I just want to show a few examples of actually using the current API to make sure you're good with it:

so this:
```
        message
            .edns_mut()
            .set_max_payload(MAX_PAYLOAD_LEN)
            .set_version(0);
```

becomes this (substitute `edns_mut` for the new method name)
```
  if message.edns_mut().is_none() {
            message.set_edns(Edns::new());
        }
        message
            .edns_mut()
            .as_mut()
            .map(|edns| edns.set_max_payload(MAX_PAYLOAD_LEN).set_version(0));
```


and inserting an option:
```
 if let Some(edns) = message.edns_mut().as_mut() {
        edns.options_mut().insert(EdnsOption::Unknown(EdnsCode::Subnet.into(),[
            0, 1, 16, 0, 1, 2, ].to_vec()));
     }
```

using `.map` will throw a clippy warning, so missing the combinator-style is a bit of a bummer.

Reply:
You could also use this:
```
 message
            .extensions_mut()
            .get_or_insert(Edns::new())
           
```

We could set the deprecation message on `edns_mut` to this as it's equivalent, I think

Reply:
I like this a lot more personally. Thanks for making these changes. It's too bad `get_or_insert_default` isn't stable yet on Option.

I think it might actually be better to use this construct though: `get_or_insert_with(Edns::new)` as that would delay the Edns construction to only when it doesn't exist.

Reply:
Ah! I was looking for the lazily constructed variant but missed that. Yes, agreed!